### ISSUE TITLE
Multiple ACF blocks Failed to load resource: the server responded with a status of 503

### ISSUE THREAD

**Issue by amitbiswas06** [AdvancedCustomFields/acf#269]
I am here with a heart breaking situation and requires your help. I have been working on my new WP theme for the last 5 months. I am using ACF pro 5.8.7 and developed 18~20 custom guten blocks with it. I have tested and found no issues in my local host which is a wamp server.

But since I moved the site to my staging web server, I am getting **admin-ajax.php errors(503)** in it. So the page with acf blocks is not loading properly in wp admin. However the front end site is working nicely.

My custom blocks are having moderate fields that requires for them. I am really hopeful that you can help me.

Thanks,
Amit Biswas

**Comment by elliotcondon** [comment]
Hi @amitbiswas06 

Sorry to hear about the issue.

I suspect this issue may be due to the number of HTTP requests made during the admin page load to render ACF's dynamic block previews.

Using your browser's inspector tools, can you please review the number of HTTP AJAX requests fired on page load when editing your page/post?

It will be very beneficial to also initiate a conversation with your web host to discuss the HTTP requests, and better understand why the server is returning 503 errors.

**Comment by amitbiswas06** [comment]
Dear Elliot,

I have checked the HTTP requests. There are about 20 HTTP AJAX request at
the time I am opening the edit page. Literally 1 request per ACF block I
guess. But that's really small isn't it? I am attaching a snapshot for your
reference. Please have a look.

There is no request made for native blocks though. I also have other sites
that have tons of ACF meta fields which are loading with no time almost.
But this time ACF block having the issue.

I am on a shared web hosting with almost no issues for my other heavy
sites. My server also configured with -

max_execution_time = 1200
max_input_time = 600
max_input_vars = 5000
memory_limit = 256M
max_file_uploads = 100
upload_max_filesize = 100M
post_max_size = 100M

Apart from that, my blocks are not too heavy, some of them have repeater
field or taxonomy field or image field or simple text field etc.

It is also noted that when I am clicking a ACF block and toggling the
edit/preview, after couple of second the block loads up. But that is not
good as I am going to sell my theme at Worlds most popular marketplace.
This will hang me down.

Please help me if you can. I can provide you a test server if needed.

Thanks,
Amit

**Comment by elliotcondon** [comment]
Hi @amitbiswas06 

Thanks for the reply.

Q1.
> Literally 1 request per ACF block I guess. But that's really small isn't it?

This is how dynamic blocks work in the Gutenberg API. Each dynamic block is rendered via an AJAX request to fetch the HTML via PHP. It is possible that we could "pre-load" the HTML on initial page load for optimal performance which I'll consider in future dev.

Q2. 
> It is also noted that when I am clicking a ACF block and toggling the edit/preview, after couple of second the block loads up.

To determine if the "slowness problem" is due to the actual fields, please register a new block without any fields assigned to it. Provide a callback function/template which simply outputs the text "Hello world". How long does it take to render this block when clicked on?

Q3.
> But that is not good as I am going to sell my theme at Worlds most popular marketplace. This will hang me down.

Correct. I would not advise it. Dynamic blocks are great for sites that require a PHP friendly environment for block development, but they will never be able to match the performance of a native static block.

Hope that helps.

**Comment by amitbiswas06** [comment]
> To determine if the "slowness problem" is due to the actual fields, please register a new block without any fields assigned to it. Provide a callback function/template which simply outputs the text "Hello world". How long does it take to render this block when clicked on?

I have tested this now. Just created a simple "Hello World" block without any ACF field. Then I have used the block 16 times in a new page. Then saved it and reopened it in admin. The result is same. admin-ajax.php 503 error same as before.

Can you advise me what I should do now? Should I switch to native block development for the simple blocks and remain in ACF block for the complex ones that have taxonomy/postObject etc?

But the issue will bug me and I guess lots of other devs too. Your advise is more important in this situation. I already have invested 4-5 months on the project and now I found myself alone in the desert without water. There is no warnings at ACF documentation for the block. Otherwise I would have planned my project different way.

Please help if possible.

Thanks,
Amit

**Comment by nick6352683** [comment]
I hope this helps, even though I have no simple solution for you here.

I have created more than 50 ACF Blocks, with some of them pretty heavy on settings (fields), some more than 30... For testing I have created moderate size pages, with about 40-50 blocks without any issues.

Just like you, ,my localserver is xampp, but my live server is a crappy/weak Godaddy shared hosting. The only difference between the 2 environments, is that in the Godaddy shared hosting the blocks load with a 2-3 second delay in the backsite, due to the facts that ACF blocks are dynamic and the actual hosting is really bad ! So I'm not experiencing any of the problems that you are reporting nor anyone else (that I know of). BTW, on the front end things load instantly, like a VPS or even a dedicated server thanks to a good cache plugin I have in place.

That said, my only advice I can provide you right now is to change the hosting company. Error 503 is a server error, so I don't expect for you to get different results by changing browsers of even devices.. . I can with almost certainly blame this on your hosting...

Also, there is no need to look at your code and see how you set and configured things, since the blocks are working in your localserver environment. I keep coming to your hosting server, every time I'm thinking how to resolve this, and at this point, probably the only people who can really help you  are the hosting people... Muck easier to dump them and get better hosting (if I were you).

**Comment by elliotcondon** [comment]
@amitbiswas06 

Thank-you for testing the simple block. These results do point towards a server level issue due to the number of simultaneous AJAX requests.

I'll add this to out to-do list and perform some testing with a larger number of dynamic blocks during page load. I can already think of a few solutions to avoid the `n x blocks` requests.

As @nick6352683 points out, it would be wise to test the issue on another web host. If this is possible, please share with me your results.

If your project allows the time to invest in "native" block development, please do so. Creating native blocks will remove all kinds of limitations found in our PHP framework, but will also require much more time.

Hope this helps.

**Comment by amitbiswas06** [comment]
@elliotcondon @nick6352683 I have tested my theme at a different server and there I found no issues with the admin-ajax.php POST 503 errors. As this is a relief for me though I really appreciate Elliot's thinking of solutions to avoid `n x blocks` requests. I am looking forward to that. 

As I don't have control over the user's/customer's choice for the hosting, I really need to think and modify my project wisely. A mix solution with **ACF dynamic blocks + Custom Native Blocks + Page templates with ACF Flexible fields** would be good I hope.

What you think @elliotcondon ?

Thank you so much to all of your valuable time.

**Comment by elliotcondon** [comment]
@amitbiswas06  Without knowing the size/scope of your editing experience, I can't make any meaningful decisions for you. Please consider all your options and do what is best for your specific project.

I'll be sure to post back here once I investigate the idea of pre-rendering the blocks to avoid multiple AJAX calls.

**Comment by amitbiswas06** [comment]
@elliotcondon Today I started using the new hosting that I have had tested once and reported that no admin-ajax.php errors are there, unfortunately I am getting same errors now while using the new hosting. Just thought you should get this latest info.

**Comment by elliotcondon** [comment]
Thanks for the update.

**Comment by despecial** [comment]
Unfortunately, I have the same experience on my local machine and there are not too many blocks. maybe 15-20 on the page. but it loads for half a minute. way too long

**Comment by elliotcondon** [comment]
Thanks for the additional info.

**Comment by Log1x** [comment]
> It is possible that we could "pre-load" the HTML on initial page load for optimal performance which I'll consider in future dev.

I really feel like this should be a priority and perhaps split off into a separate issue if it hasn't been already.

With ACF 5.9 having Innerblock support – it is quickly turning into a performance nightmare...

**Comment by elliotcondon** [comment]
Hi @Log1x - How many ACF Blocks are you adding to the page?

**Comment by joelschlotterer** [comment]
I am in a similar situation. My team has spent months building out ACF based blocks to now discover that in it is completely unusable when editing an existing page (502/504 errors from too many requests). On a page with 35 blocks, it took me 16 minutes to make a text change to a block built with ACF. The page I'm currently scrambling to examine has 30 ACF blocks mixed in with some WP blocks. We had to max out an entire GCP box to get a client site revisions done in a timely manor. I really hope to hear some kind info on a plan for addressing this.

**Comment by elliotcondon** [comment]
Hi @joelschlotterer 

Thanks for the comment. I'm sorry to hear how badly this AJAX request performance issue is effecting your project. This issue is the crux of our PHP based framework, but it is definitely something that can be fixed.

The issue is simply due to the way that dynamic blocks are loaded in the editor - via AJAX. This isn't an issue on the front end (thank goodness), but is a serious issues for content editors.

Our idea is to render all the blocks in the backend, similar to how they are rendered on the front end (without individual AJAX requests). The reason why your site is timing out - or taking multiple minutes to load is most likely a protection layer on your server that is preventing it from crashing on too many simultaneous requests.

I'll hope to have some solutions for you shortly :)

**Comment by Log1x** [comment]
Sorry for the late reply.

The block count comes out to be somewhat similar to @joelschlotterer 

In my case, it is about 8-9 ACF blocks that each have 4-5 other ACF blocks inside of their InnerContent. (e.g. a How To block with multiple "Step" blocks inside)

**Comment by fabiofolgori** [comment]
I have the same problem. I have 10 blocks and it takes30 seconds to load. Sometimes the request goes in 503 and they do not load themselves. In the frontend everything is ok instead

I hope a quick bugfix

thanks

**Comment by hatchetaustralia** [comment]
We have the same issue, the site was running on PHP 7.4. Since downgrading to PHP 7.3 the issue has resolved itself.

**Comment by artemsivak** [comment]
Same issue here. All our multisite project based on acf blocks. And it's a nightmare to edit page with a lot of blocks. It takes a lot of time to edit something

**Comment by elliotcondon** [comment]
Hi all 👋. I'm very pleased to announce we have a solution for this issue, and are pushing hard to get a release out tomorrow 🎉🎉!

The solution we ended up with is very elegant but requires testing. We understand the time sensitive nature of this issue, and as a result, have decided to treat this fix as "experimental" for version 5.9.2 with the goal of lifting this "flag" in version 5.9.3.

What does this mean exactly? Simply put, this allows us to roll out the fix to developers willing to test without the risk of effecting any other live sites. After updating to version 5.9.2, please add the following code to your `wp-config.php` to enable the fix:
```
define('ACF_EXPERIMENTAL_PRELOAD_BLOCKS', true);
```

Please post your result here on GitHub. We're very excited about tomorrow's release - see you then!

**Comment by palmiak** [comment]
@elliotcondon I did some quick tests and it looks great. I had a problem with many ACF Blocks being loaded at once on one cheap hosting. With PRELOAD enabled it works blazing fast with no errors.

**Comment by Hackima** [comment]
What a great new ! @elliotcondon , it seems that only preview mode is going to be preloaded. Any chance that edit mode will be preloaded too ?

**Comment by elliotcondon** [comment]
Hi @Hackima - That might be a little tricky, but let me consider it for a future update 👍

**Comment by justinkruit** [comment]
Just tried the experimental feature and I'm absolutely loving it. Haven't noticed any problems with it.

**Comment by joelschlotterer** [comment]
**First, thank you so much for making this fix a priority!**  
We are now testing in our environments and the preview of the blocks is great! I agree with @Hackima that it would be ideal if both the preview and the fields were preloaded (if possible). Having to encourage our users to keep the blocks in preview mode until they want to edit them isn't ideal, but this update does make our blocks usable in the admin again. So again thank you so much for getting this out quickly!
Will the final version of this update still require a line in the WP config?
Is there a roadmap/ time frame you're looking at to release 5.9.3 with the final fix?

**Comment by joshuafredrickson** [comment]
The experimental flag reduced a page with 12 blocks from about 30 seconds load time down to about 5.

This has changed my life. 🙏

**Comment by JiveDig** [comment]
Another +1 here. I don't really need edit mode for our stuff, personally.

But our theme that has 2 blocks with 30+ fields is loading lightning fast now. Great work!

**Comment by Log1x** [comment]
+1 – it's like magic. Thanks!

**Comment by thomasjulienalain** [comment]
Love it ! Thank you.

**Comment by arjendejong12** [comment]
@elliotcondon Did this constant get removed in 5.9.3? I have it enabled but it's not preloading anything, unlike 5.9.2.

**Comment by elliotcondon** [comment]
@arjendejong12 No. The constant has not been removed in 5.9.3. Can you please elaborate more on the issue?

**Comment by virgo79** [comment]
I can't see any valuable change too. 
Checking the browser inspector I see that Block preload always rely on ajax requests.

Tested with both ACF 5.9.2 and 5.9.3.
Below ACF (5.9.2) is network enabled on WP 5.5.1 multisite.
![acf-preload](https://user-images.githubusercontent.com/6567583/99378525-ceea5a00-28c7-11eb-9cff-be65ebb5d9e0.gif)

**Comment by despecial** [comment]
Did you add this to your config? @virgo79 
`define('ACF_EXPERIMENTAL_PRELOAD_BLOCKS', true);
`

**Comment by virgo79** [comment]
@despecial yes.
I've also updated to wp 5.5.3.

Below is a more details screenshot with ajax requests panel.

![acf-preload-2](https://user-images.githubusercontent.com/6567583/99382611-13c4bf80-28cd-11eb-9644-95be50f1ab17.gif)

**Comment by amitbiswas06** [comment]
@virgo79 It seems the nature of the problem is not understood properly. Please refer the actual question followed by @elliotcondon and my discussion.

**Comment by elliotcondon** [comment]
Hi @virgo79 - The demonstration in your screen-record is not something that the `ACF_EXPERIMENTAL_PRELOAD_BLOCKS` constant will change. Due to the dynamic nature of ACF Blocks, there will always be an AJAX call fired after making a "change" to field values.

The `ACF_EXPERIMENTAL_PRELOAD_BLOCKS` constant is intended to help speed up the initial page load by preloading any previewed that will be rendered on the edit screen.

**Comment by joelschlotterer** [comment]
@elliotcondon so is the plan to leave the ACF_EXPERIMENTAL_PRELOAD_BLOCKS constant as a requirement for this functionality?
We are trying to decide if we need to script this into our dev ops process for setting up and maintaining sites.
Is there a release coming where this functionality is no longer considered experimental?

**Comment by elliotcondon** [comment]
Hi @joelschlotterer. This constant will only be required whilst we consider the functionality "experimental". With enough feedback and bug fixes, we will merge this into core without the need to define any constant 👍.

**Comment by virgo79** [comment]
> Hi @virgo79 - The demonstration in your screen-record is not something that the `ACF_EXPERIMENTAL_PRELOAD_BLOCKS` constant will change. Due to the dynamic nature of ACF Blocks, there will always be an AJAX call fired after making a "change" to field values.
> 
> The `ACF_EXPERIMENTAL_PRELOAD_BLOCKS` constant is intended to help speed up the initial page load by preloading any previewed that will be rendered on the edit screen.

@elliotcondon thanks for the clarification!

**Comment by Fu-San** [comment]
@elliotcondon The new experimental constant ACF_EXPERIMENTAL_PRELOAD_BLOCKS fixes one of the issues I was facing in the backend. Some contents weren't showing in the frontend but it worked in the backend properly. 
For example
<h2><?php echo get_the_title(); ?> Overview</h2> didn't show in backend editor but worked correctly in frontend. This is fixed.

But the below function still doesn't work in the backend. It works in frontend as usual.
<img src="<?php echo get_the_post_thumbnail_url( null, 'full' ); ?>" class="character-icon" alt="Traveler (Anemo)">

It appears like this in the editor
![Screenshot 2020-11-21 091145](https://user-images.githubusercontent.com/35169687/99866253-dc5e5780-2bd9-11eb-9673-1206fadf5a33.jpg)

**Comment by elliotcondon** [comment]
Hi @Fu-San. Thanks for the comment. Please keep in mind that some WP functions such as `get_the_title()` and `get_the_post_thumbnail_url()` refer to a global `$post` object if the `$post_id` variable is not provided (or in your case set to null). During the AJAX request to render the block preview, that global `$post` object is not setup, which explains why those functions are not running as intended.

The soution is simple, either setup the gobal `$post` object manually, or use the provided `$post_id` parameter in any `get_the_title()` or `get_the_post_thumbnail_url()` calls 👍.

**Comment by gallivanmedia** [comment]
> Hi @joelschlotterer. This constant will only be required whilst we consider the functionality "experimental". With enough feedback and bug fixes, we will merge this into core without the need to define any constant 👍.

I always have my clients see the "preview" first... so, this "experimental" fix with the ACF_EXPERIMENTAL_PRELOAD_BLOCKS 
 constant is exactly what I need. From there, I don't even mind them waiting for the individual block editors to load. THANK YOU for the quick solution.

**Comment by joelschlotterer** [comment]
So still loving this fix. I did notice that it doesn't preload the preview for Reusable blocks. I don't know if that is something you can even control, just wanted to note it though.
Again, without this fix we would have had to start over so thank you for taking this seriously and we appreciate your efforts!

**Comment by kmcaloon** [comment]
@elliotcondon appreciate you being so responsive. I might be missing something, but since admin-ajax tends to be pretty resource intensive, is there a reason why the rest api couldn’t be leveraged here?

**Comment by elliotcondon** [comment]
Hi @kmcaloon. In this use-case, the difference between admin-ajax and a custom REST API endpoint would be negligible. The issue here is simply due to multiple asynchronous requests hitting the server at once, which is solved completely via a preloading solution (like the one we are working on) or a JS solution to delay the AJAX request (similar to lazyloading). 

We're confident this won't be am issue for much longer, but will have to wait until the new year until anything get's released :)

**Comment by kmcaloon** [comment]
Thanks @elliotcondon. Totally understood on the timing. As far as admin-ajax, my first concern when seeing the network requests while interacting with the blocks was actually the impact on the server it would have while multiple content editors edit the site throughout the day. Unfortunately I don’t think preloading would help here.

I’m assuming custom endpoints wouldn’t help here because either way acf would need to load a ton of WP core files anyway?

**Comment by smerriman** [comment]
By default, when I create an ACF block in preview mode that has links in it, clicking on a link inside the block in the editor redirects the browser, rather than allowing me to edit the block.

I've been resolving this by adding:

`.acf-block-preview a {pointer-events:none;}`

to my editor styles. However, with the experimental constant, the acf-block-preview class isn't added.

Can this class be retained during preload? (And/or, is there a better way of preventing clicks in the first place?)

**Comment by JiveDig** [comment]
I do the same, but scoped to my specific block by the container html class. You probably should do similar too, so it only applies to the blocks you want it to. It still works for me with the constant added.

**Comment by smerriman** [comment]
That does work, but given you *never* want someone who clicks on an ACF preview to be taken away from the editor, it's better to have a generic class rather than have to add extra code for each new block. Given it's still a preview, the lack of a preview class seems accidental.

**Comment by JiveDig** [comment]
I have a block where a link click opens a tab, or launches a modal, via :target. That css would break my block in the backend. It's a side convo to your actual question, so my apologies there, just confirming that a link is not always a text link that needs disabling.

**Comment by jonathanrich1986** [comment]
Hi, just spotted a bug when enabling the `ACF_EXPERIMENTAL_PRELOAD_BLOCKS ` constant. My blocks which have `<InnerBlocks />`no longer appear in the editor (still works on the front end). Thanks.

**Comment by Log1x** [comment]
> Hi, just spotted a bug when enabling the `ACF_EXPERIMENTAL_PRELOAD_BLOCKS ` constant. My blocks which have `<InnerBlocks />`no longer appear in the editor (still works on the front end). Thanks.

hmm, I can't reproduce this.

**Comment by JiveDig** [comment]
Same. InnerBlocks works for me with or without the constant.

**Comment by elliotcondon** [comment]
@jonathanrich1986 - When possible, can you please put together a basic code example to demonstrate the issue? I'll setup this block in my local environment and see if I can diagnose the problem. It would be great if we could open a new GitHub ticket to work in too 👍.

@JiveDig @Log1x - Thanks for confirming 👍.

**Comment by tolnem** [comment]
I've been able to reproduce the issue where InnerBlocks don't appear with `ACF_EXPERIMENTAL_PRELOAD_BLOCKS` enabled. It happens for me if there is not a single wrapper around the contents of the block.

So as an example, this template will only render the h1 in the admin part:
`<h1>Testing</h1><div><InnerBlocks /></div>`

While this will render both the h1 and the innerblocks:
`<div><h1>Testing</h1><div><InnerBlocks /></div></div>`

**Comment by Snelwebcenter** [comment]
Voor de Nederlanders onder ons. Indien je last hebt van trage ACF Blokken in de back-end dan is dit de oplossing.

**Comment by waynemcmahon** [comment]
@elliotcondon thanks for this. I've added _define('ACF_EXPERIMENTAL_PRELOAD_BLOCKS', true);_ to wp-config.php but I am still receiving 504 errors in the backend when there are a lot of block on the page. 

![image](https://user-images.githubusercontent.com/8567038/121891520-72db9d00-cd13-11eb-9159-1145dfbc9837.png)
![image](https://user-images.githubusercontent.com/8567038/121891560-8129b900-cd13-11eb-93a9-6aeb1e074160.png)

**Comment by Snelwebcenter** [comment]
> @elliotcondon thanks for this. I've added _define('ACF_EXPERIMENTAL_PRELOAD_BLOCKS', true);_ to wp-config.php but I am still receiving 504 errors in the backend when there are a lot of block on the page.
> 
> ![image](https://user-images.githubusercontent.com/8567038/121891520-72db9d00-cd13-11eb-9159-1145dfbc9837.png)
> ![image](https://user-images.githubusercontent.com/8567038/121891560-8129b900-cd13-11eb-93a9-6aeb1e074160.png)

What is the amount of memories in WP?

**Comment by waynemcmahon** [comment]
> > @elliotcondon thanks for this. I've added _define('ACF_EXPERIMENTAL_PRELOAD_BLOCKS', true);_ to wp-config.php but I am still receiving 504 errors in the backend when there are a lot of block on the page.
> > ![image](https://user-images.githubusercontent.com/8567038/121891520-72db9d00-cd13-11eb-9159-1145dfbc9837.png)
> > ![image](https://user-images.githubusercontent.com/8567038/121891560-8129b900-cd13-11eb-93a9-6aeb1e074160.png)
> 
> What is the amount of memories in WP?

It is 512MB currently
![image](https://user-images.githubusercontent.com/8567038/121895421-1a5ace80-cd18-11eb-87eb-40e766cfe83d.png)

**Comment by elliotcondon** [comment]
Hi @waynemcmahon Thanks for the report. 

Some house-cleaning first: Can you please remove the two duplicate replies above this one?

As for the issue, it looks as if the constant is not having any effect on the page load - hence the spinning wheels displayed in each block and the large number of AJAX calls logged in the browser. Can you please confirm which version of ACF PRO you are using on this website?

**Comment by waynemcmahon** [comment]
Hi @elliotcondon . Thanks for you response and I've deleted my duplicate post.

I am running version 5.9.5 currently. Adding the constant has had no measurable affect so that I can see

**Comment by elliotcondon** [comment]
@waynemcmahon Interesting. Would you be available to perform some debugging?
1. Check file `pro/blocks.php` and log the `$preloaded_blocks` value. This should contain all HTML of your blocks during page load. Does it?
2. In your browser, check the JS data for `acf.get('preloadedBlocks')`. Does this contain the same data?
3. Can you spin up a fresh WP install without any other plugins to test if this issue is connected to a plugin compatibility issue?

**Comment by smerriman** [comment]
Just jumping in to say that the one block that *did* load in the screenshot was a block in edit mode. If all of his blocks were in edit mode as I suspect, then the constant does nothing.

+1 to the earlier request about supporting preload for blocks in edit mode.

**Comment by robjimgreen** [comment]
I'm also getting these admin-ajax.php 503 errors – All our custom ACF blocks are locked to edit mode.

Are you able to add the preload support for edit mode please?

**Comment by robjimgreen** [comment]
@bradt @elliotcondon – I understand ACF was recently acquired by Delicious Brains, are these issues still being actively monitored during the transition?

**Comment by mattgrshaw** [comment]
@robjimgreen

Yep, we're still monitoring these issues and plan to move the experimental preload blocks functionality into core in the next release. I've made a note to check into supporting edit mode for this release as well 👍

**Comment by justinkruit** [comment]
Hey @mattgrshaw , that's great news.

Is there any insight you could give about how far this next release is? I always love having new features to play around with, and looking forward to what Delicious Brains is gonna bring to the table

**Comment by mattgrshaw** [comment]
@justinkruit it's a fairly big release with a lot of things to test, so it's likely a few weeks away at this point. We don't have an exact ETA just yet.

**Comment by robjimgreen** [comment]
@mattgrshaw I've just been speaking with our hosting provider and apparently having an option to rate-limit the ajax calls could also fix this – I've just tried to use the WordPress Heartbeat Control plugin but I don't think it affects ACF.

**Comment by mattgrshaw** [comment]
@robjimgreen I think rate-limiting probably isn't ideal here, you'd either see a slowdown with ACF blocks or some of them might not load at all.

It does sound like preloading blocks set to edit mode is the way to go here, and I've raised a PR for that on our private repo for pro. Maybe once that's done our initial review process, you could kick the tires on it as well and see if it fixes the issue for you?

**Comment by robjimgreen** [comment]
@mattgrshaw No problem at all & thanks for the info – More than happy to test this out :)

**Comment by pmascis** [comment]
@mattgrshaw 

Hey there,
5.9 with define('ACF_EXPERIMENTAL_PRELOAD_BLOCKS', true);
Worked well for me, but 5.10 now causes errors when duplicating blocks.

Duplicating a block and editing the new blocks content updates the original block and more mixed up functionality between the blocks. Duplicated blocks are have a delayed load when the page is loaded in wp-admin.

This problem has made duplicated/duplicating blocks unusable. Furthermore this error means, I can't update ACF on any of my sites using Gutenberg.

Has anyone else encountered similar?

I made a support ticket on the ACF site, but I thought I'd mention here anyway. Let me know if you need more information.

**Comment by mattgrshaw** [comment]
@pmascis Yep, we're aware of this bug and found a fix, this will be rolled out in 5.10.2 this coming week.

**Comment by SergiArias** [comment]
I still have this problem with 5.12.2... Was it really solved?

**Comment by lgladdy** [comment]
> I still have this problem with 5.12.2... Was it really solved?

We shipped block preloading, but due to a breaking change made by the Gutenberg team in WordPress 5.9, had to disable edit mode block preloading again in ACF 5.12 until we can restructure the way edit mode preloading works. We're hoping to re-enable this in our next release.

**Comment by mike-sheppard** [comment]
hey, @lgladdy hate to chase as I know you guys are super busy + appreciate all the hard work going into all the ACF updates. But do you have an ETA on preloading being re-enabled? One of our partner's sites is grinding to a halt while editing

**Comment by lgladdy** [comment]
@mike-sheppard Preloading should only be disabled for blocks saved in edit mode, but if you've got loads of blocks like that then I understand this is frustrating!

We're targeting re-enabling of this for our next release, but that's likely a couple of months away. We'll try and get this out in a beta in the coming weeks though.

WordPress changed the scope in which internal REST preloads take place in WordPress 5.9, this means any assets that get enqueued are deemed satisfied by the REST preload, but then aren't enqueued for the actual page - meaning a bunch of ACF's JavaScripts aren't loaded correctly in the block editor - we're trying to overcome that issue by detecting when an internal REST preload happens, but WordPress don't offer any way to determine that automatically, so it's a significant challenge for us to overcome - that's why we weren't able to roll out a quick fix in one of the minor releases for ACF 5.12

**Comment by luicephillips** [comment]
I have tried with all above solutions, but still the issue is not resolved, can anybody please help me

**Comment by michelmany** [comment]
@lgladdy, thanks for your detailed reply. JFYI, I am still getting this error with v. ACF PRO 5.12.2.
It works fine locally but not in the client's hosting.
Thank you.

**Comment by lgladdy** [comment]
@michelmany Our current preview build (#654) will resolve this issue, but it's currently in alpha so not recommended for production sites.

**Comment by Rickyc81** [comment]
Thanks.. I've same issue in loading of multiple block in edit mode.. :(
I switch it in preview mode for remedy the problem momentarily.. 
when is the release scheduled? 

Thanks!!

**Comment by davidwebca** [comment]
@lgladdy Sorry for the ping. Is the preview build you're talking about 6.0.0-RC3?

**Comment by Rai-burger** [comment]
6.2.9 here and experiencing the same issue ☹
![image](https://github.com/AdvancedCustomFields/acf/assets/85535429/3ddf2ee3-5d7a-46a5-b875-952040161a09)
![image](https://github.com/AdvancedCustomFields/acf/assets/85535429/502441d4-ae7f-41c2-a114-9e5cbc3495f9)

**Comment by lgladdy** [comment]
@Rai-burger It would seem block preloading isn't working for you here, unless you've changed every block in the code editor (in which case, you'd expect this to happen)

We're getting close to releasing ACF 6.3.0-beta1, I'll leave in a bunch of preload debugging in that so you can see what's going on. We may have fixed the preloading bug in that anyway (it seems WP 6.5 changed the way some blocks are initialised, meaning you could miss the preload on the second load)
